Shared/Headers/FocusEventListener.hpp

```diff
@@ -23,44 +23,38 @@ class FocusEventListener {
     std::unique_ptr<zmq::socket_t> _socketSUB = std::make_unique<zmq::socket_t>(*FocusSocket::Context, ZMQ_SUB);
     std::atomic<bool> _isRunning;
 
-    void RunReceive(zmq::socket_t *socketSUB, const std::function<void(TPayload &)> onMessage) {
+    void RunReceive() {
         while (_isRunning) {
-            zmq::message_t msg;
-            if (socketSUB->recv(&msg)) {
-                if (!socketSUB->recv(&msg))
-                    continue;
-                std::string payload = std::string(static_cast<char *>(msg.data()), msg.size());
+            zmq::multipart_t msg;
+            if (msg.recv(*_socketSUB)) {
+                std::string payload = std::string(static_cast<char *>(msg.at(1).data()), msg.at(1).size());
                 Focus::Event event;
                 if (!event.ParseFromString(payload))
                     continue;
-                onMessage(event);
+                _onMessage(event);
             }
         }
     }
 
-    void RunReceiveEnvelope(zmq::socket_t *socketSUB, const std::function<void(TPayload &)> onMessage) {
+    void RunReceiveEnvelope() {
         while (_isRunning) {
-            zmq::message_t msg;
-            if (socketSUB->recv(&msg)) {
-                if (!socketSUB->recv(&msg))
-                    continue;
-                std::string payload = std::string(static_cast<char *>(msg.data()), msg.size());
+            zmq::multipart_t msg;
+            if (msg.recv(*_socketSUB)) {
+                std::string payload = std::string(static_cast<char *>(msg.at(1).data()), msg.at(1).size());
```

Comment:
As a rule of thumb, try to avoid copy assignments when constructing objects.
Although it does not matter here since `std::string` has a move operator